tool/clean/clean_log_group/clean_log_group.go

```diff
@@ -0,0 +1,280 @@
+
+	// Compute cutoff times
+	cutoffTimes := calculateCutoffTimes()
+
+	logger.Printf("🔍 Searching for CloudWatch Log Groups older than %d days AND inactive for %d days in %s region\n",
+		cfg.thresholdDays, cfg.inactiveDays, awsCfg.Region)
+
+	// Delete old log groups
+	deletedGroups := deleteOldLogGroups(ctx, client, cutoffTimes)
+	logger.Printf("Total log groups deleted: %d", len(deletedGroups))
+}
+
+type cutoffTimes struct {
+	creation int64
+	inactive int64
+}
+
+func calculateCutoffTimes() cutoffTimes {
+	return cutoffTimes{
+		creation: time.Now().AddDate(0, 0, -cfg.thresholdDays).Unix() * 1000,
+		inactive: time.Now().AddDate(0, 0, -cfg.inactiveDays).Unix() * 1000,
+	}
+}
+
+func loadAWSConfig(ctx context.Context) (aws.Config, error) {
+	cfg, err := config.LoadDefaultConfig(ctx)
+	if err != nil {
+		return aws.Config{}, fmt.Errorf("loading AWS config: %w", err)
+	}
+	cfg.RetryMode = aws.RetryModeAdaptive
+	return cfg, nil
+}
+
+func deleteOldLogGroups(ctx context.Context, client *cloudwatchlogs.Client, times cutoffTimes) []string {
+	var (
+		wg                sync.WaitGroup
+		mutex             sync.Mutex
+		logGroupsToDelete []string
+		logGroupChan      = make(chan *types.LogGroup, 500)
+	)
+
+	// Start worker pool
+	logger.Printf("👷 Creating %d workers\n", cfg.numWorkers)
+	for i := 0; i < cfg.numWorkers; i++ {
+		wg.Add(1)
+		go processLogGroup(ctx, client, logGroupChan, &wg, &mutex, &logGroupsToDelete, times, i)
+	}
+
+	// Process log groups in batches
+	if err := fetchAndProcessLogGroups(ctx, client, logGroupChan, &logGroupsToDelete, &mutex); err != nil {
+		logger.Printf("Error processing log groups: %v", err)
+	}
+
+	close(logGroupChan)
+	wg.Wait()
+
+	return logGroupsToDelete
+}
+
+func processLogGroup(ctx context.Context, client *cloudwatchlogs.Client, logGroupChan <-chan *types.LogGroup,
+	wg *sync.WaitGroup, mutex *sync.Mutex, logGroupsToDelete *[]string, times cutoffTimes, workerID int) {
+	defer wg.Done()
+
+	for logGroup := range logGroupChan {
+		if err := handleLogGroup(ctx, client, logGroup, mutex, logGroupsToDelete, times, workerID); err != nil {
+			logger.Printf("Worker %d: Error processing log group: %v", workerID, err)
+		}
+	}
+}
+
+func handleLogGroup(ctx context.Context, client *cloudwatchlogs.Client, logGroup *types.LogGroup,
+	mutex *sync.Mutex, logGroupsToDelete *[]string, times cutoffTimes, workerID int) error {
+
+	if logGroup.CreationTime == nil {
+		return fmt.Errorf("log group has no creation time: %v", logGroup)
+	}
+
+	logGroupName := *logGroup.LogGroupName
+	creationTime := *logGroup.CreationTime
+
+	if creationTime >= times.creation {
+		return nil
+	}
+
+	lastLogTime := getLastLogEventTime(ctx, client, logGroupName)
+	if lastLogTime == 0 {
+		return nil
+	}
+
+	if lastLogTime < times.inactive {
+		logger.Printf("🚨 Worker: %d| Old & Inactive Log Group: %s (Created: %v, Last Event: %v)\n",
+			workerID, logGroupName, time.Unix(creationTime/1000, 0), time.Unix(lastLogTime/1000, 0))
```

Comment:
nit: Can just use `time.UnixMilli`.
```suggestion
			workerID, logGroupName, time.UnixMilli(creationTime), time.UnixMilli(lastLogTime))
```